helper/keysutil/policy.go

```diff
@@ -366,7 +366,29 @@ func (p *Policy) handleArchiving(ctx context.Context, storage logical.Storage) e
 	return nil
 }
 
-func (p *Policy) Persist(ctx context.Context, storage logical.Storage) error {
+func (p *Policy) Persist(ctx context.Context, storage logical.Storage) (retErr error) {
+	// Other functions will take care of restoring other values; this is just
+	// responsible for archiving and keys since the archive function can modify
+	// keys. At the moment one of the other functions calling persist will also
+	// roll back keys, but better safe than sorry and this doesn't happen
+	// enough to worry about the speed tradeoff.
+	priorArchiveVersion := p.ArchiveVersion
+	priorKeys := keyEntryMap(nil)
```

Comment:
Could this use `copystructure.Copy(p.Keys)` so that we can just do `p.Keys = priorKeys.(keyEntryMap)` similar do what is being done in `Upgrade()`?

Reply:
Discussed internally, removed copystructure instead.